src/praisonai-bot/praisonai_bot/gateway/pairing.py

```
@@ -40,44 +40,123 @@ def _get_secret() -> str:
+def _restrict_windows_acl(secret_path: str) -> bool:
+    """Best-effort restrict a file's ACL to the current user on Windows.
+
+    Returns ``True`` when the ACL was applied, ``False`` otherwise. Never
+    raises — callers treat failure as non-fatal.
+    """
+    try:
+        import getpass
+        import subprocess
+
+        user = os.environ.get("USERNAME") or getpass.getuser()
+        result = subprocess.run(
```

Ruff (0.15.20)

[error] `subprocess` call: check for execution of untrusted input

(S603)